common/buildcraft/builders/snapshot/Blueprint.java

```diff
@@ -137,19 +139,32 @@ public EnumSnapshotType getType() {
         public final Map<ISchematicEntity<?>, List<ItemStack>> entitiesRequiredItems = new HashMap<>();
         public final Map<ISchematicEntity<?>, List<FluidStack>> entitiesRequiredFluids = new HashMap<>();
 
+        private boolean finishedComputing = false;
+        private final Future<BlueprintCalculator.BuildingInfoData> future;
+
         public BuildingInfo(BlockPos basePos, Rotation rotation) {
             this.basePos = basePos;
             this.rotation = rotation;
-            Pair<List<ItemStack>[][][], List<FluidStack>[][][]> required = SchematicBlockManager.computeRequired(
-                getSnapshot());
-            Pair<List<List<ItemStack>>, List<List<FluidStack>>> requiredEntities = SchematicEntityManager
-                .computeRequired(getSnapshot());
+            box = new Box();
+            future = WorkerThreadUtil.executeWorkTask(new BlueprintCalculator(Blueprint.this), false);
+        }
+
+        private void processData() {
+            BlueprintCalculator.BuildingInfoData infoData = null;
+            try {
+                infoData = future.get();
+            } catch (InterruptedException | ExecutionException e) {
+                //Future reported to be done and we are not canceling so this should never heapen
+                e.printStackTrace();
```

Comment:
If an error should never happen then we probably need to know about it: Please throw a ````RuntimeException```` or an ````Error```` instead of discarding the ````ExecutionException````. I'm not sure what the best behaviour is for handling  ````InterruptedException```` though.